src/hotspot/share/runtime/os.cpp

```diff
@@ -1317,6 +1318,11 @@ void os::print_location(outputStream* st, intptr_t x, bool verbose) {
   }
 #endif
 
+  // Ask if any OopStorage knows about this address.
+  if (OopStorageSet::print_containing((oop*)addr, st)) {
```

Comment:
`addr` might not be oop-aligned, in which case this cast and use might lead to UB.  I think this should
be gated on `is_aligned(addr, alignof(oop))`.

Reply:
I don't want to lose the match if the pointer is unaligned. The unaligned pointer might is still be technically _in range_ that is covered by the OopStorage. See new commit, I think we can handle it more accurately without losing this match?